### PR TITLE
fix: add `eager` as options to types and docs

### PR DESCRIPTION
**Pull request by aklkv**
related: https://github.com/ember-modifier/ember-modifier/pull/486
fixes: https://github.com/ember-modifier/ember-modifier/issues/774

### CODE REVIEW

**Review by NullVoxPopuli — CHANGES_REQUESTED** [review]
The option is not in the current version of ember-modifier -- it was in the previous major.

Only folks using v3 need to add eager false

File: README.md [hunk 1]
```diff
@@ -10,31 +10,32 @@ function-based modifiers and more complicated class-based modifiers.
 
 **NOTE:** this is the README for the v4 release. For the v3 README, see [here](https://github.com/ember-modifier/ember-modifier/blob/v3/README.md).
 
-- [Compatibility](#compatibility)
```
Comment (NullVoxPopuli) [inline README.md:13]: Why'd this change?

**Comment by NullVoxPopuli** [comment]
If you wanted to update the v3 readme, that'd be fine 🎉

**Comment by SergeAstapov** [comment]
Agree with @NullVoxPopuli, this option does not exist in v4 hence IMO this is not a legit change

**Comment by aklkv** [comment]
@SergeAstapov yeah we synced up with @NullVoxPopuli on Discord and in alignment. I think I will close this PR and when time permits add notes to v3 docs

**Pull request closed without merging**